src/app/shared/data-input/data-input.component.ts

```diff
@@ -51,12 +70,71 @@ export class DataInputComponent {
   @Output()
   public onFocus = new EventEmitter<any>();
 
+  private tempElement: HTMLElement;
   public readonly constraintType = ConstraintType;
 
+  constructor(private renderer: Renderer2, private elementRef: ElementRef) {}
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes.value && this.resizeToContent) {
+      this.recalculateWidth(this.value);
+    }
+  }
+
+  private recalculateWidth(value: any) {
+    const width = this.getWidthOfInput(value);
+    this.renderer.setStyle(this.elementRef.nativeElement, 'width', `${width}px`);
+  }
+
+  private getWidthOfInput(value: any): number {
+    if (this.constraint && this.constraint.type === ConstraintType.Boolean) {
+      return 34;
+    }
+
+    if (!this.tempElement) {
+      this.tempElement = this.createTempElement();
+      document.body.appendChild(this.tempElement);
```

Comment:
What is this temporary element good for?

Reply:
It it used for computing `input` width when `resizeToContent` mode is enabled.